src/tests/util.lua

```diff
@@ -324,94 +217,48 @@ function test_util.parse_pma_file(filename)
     return data
 end
 
--- PMA defs
-local PMA_SHADOW_STATE_LENGTH = 0x1000
-local PMA_SHADOW_PMAS_LENGTH = 0x1000
-local PMA_ROM_LENGTH = 0xF000
-local PMA_SHADOW_TLB_START = 0x20000
-local PMA_SHADOW_TLB_LENGTH = 0x6000
-local PMA_CLINT_START = 0x2000000
-local PMA_CLINT_LENGTH = 0xC0000
-local PMA_HTIF_START = 0x40008000
-local PMA_UARCH_RAM_START = 0x70000000
-local PMA_RAM_START = 0x80000000
-local PMA_PAGE_SIZE_LOG2 = 12
-local PMA_PAGE_SIZE = 1 << PMA_PAGE_SIZE_LOG2
+local PAGE_LOG2_SIZE = 12
+local PAGE_SIZE = 1 << PAGE_LOG2_SIZE
+local WORD_LOG2_SIZE = 3
 
-local function ceil_log2(x)
-    assert(x > 0, "shouldn't try to compute log of x <= 0")
-    return math.ceil(math.log(x, 2)) // 1
+local function merkle_hash(data, start, log2_size)
+    if log2_size > WORD_LOG2_SIZE then
```

Comment:
Checking if we are trying to compute merkle hash of a page of zeros can speed up this function by 7x factor here, and this speed up our tests. Ex:

```lua
local ZERO_PAGE = string.rep('\x00', PAGE_SIZE)
local function merkle_hash(data, start, log2_size)
    if log2_size == PAGE_LOG2_SIZE and data:sub(start + 1, start + (1 << log2_size)) == ZERO_PAGE then
        return zero_keccak_hash_table[PAGE_LOG2_SIZE]
    end
```